Chummer/frmCareer.cs

```diff
@@ -16313,6 +16368,295 @@ private void AddCyberwareSuite(Improvement.ImprovementSource objSource)
+
+                                if (refreshingPanel == null)
+                                    continue;
+
+                                intSustainedObjects = refreshingPanel?.Controls.Count ?? 0;
+
+
+                                SustainedSpellControl objSustainedSpellControl = new SustainedSpellControl(objSustained);
+
+                                objSustainedSpellControl.SpellDetailChanged += MakeDirtyWithCharacterUpdate;
+                                objSustainedSpellControl.UnsustainSpell += DeleteSustainedSpell;
+
+
+                                objSustainedSpellControl.Top = intSustainedObjects * objSustainedSpellControl.Height;
+
+
+                                refreshingPanel.Controls.Add(objSustainedSpellControl);
+
+                                
+                            }
+                            CharacterObject.OnPropertyChanged(nameof(CharacterObject.SustainedCollection));
+                        }
+                        break;
+                    case NotifyCollectionChangedAction.Reset:
+                        {
+                            RefreshSustainedSpells(flpSustainedSpells, flpSustainedComplexForms, flpSustainedCritterPowers);
+                        }
+                        break;
+                }
+            }
+
+            if (CharacterObject.SustainedCollection.Count == 0)
+            {
+                CharacterObject.PsycheActive = false;
+                chkPsycheActive.Visible = false;
+            }
+        }
+
+        /// <summary>
+        /// Adds a sustained spell
+        /// </summary>
+        private void AddSustainedSpell()
+        {
+            //Could be merged with AddSustainedComplex form and create an more or less universal method to add ISustainables from tre viewers. Not worth the trouble and probably not better at the moment.
+            if (treSpells.SelectedNode != null && treSpells.SelectedNode.Level > 0 && treSpells.SelectedNode.Tag is Spell objSpell)
+            {
+                SustainedSpell objNewSustainedSpell = new SustainedSpell(CharacterObject);
+
+                objNewSustainedSpell.Create(objSpell);
+
+                CharacterObject.SustainedCollection.Add(objNewSustainedSpell);
+
+                IsCharacterUpdateRequested = true;
+                IsDirty = true;
+            }
+        }
+
+        /// <summary>
+        /// Adds a sustained complex form
+        /// </summary>
+        private void AddSustainedComplexForm()
+        {
+            if (treComplexForms.SelectedNode != null && treComplexForms.SelectedNode.Level > 0 && treComplexForms.SelectedNode.Tag is ComplexForm objComplexForm)
+            {
+                SustainedComplexForm objSustained = new SustainedComplexForm(CharacterObject);
+
+                objSustained.Create(objComplexForm);
+
+                CharacterObject.SustainedCollection.Add(objSustained);
+
+                IsCharacterUpdateRequested = true;
+                IsDirty = true;
```

Comment:
These should be stuck into `SustainedSpellCollectionChanged` somehow, otherwise the program won't properly refresh itself if a sustained object is deleted outside of this method. That's why those events are set up in the first place.